contracts/SubdomainRegistrar.sol

```diff
@@ -104,18 +116,36 @@ contract SubdomainRegistrar is RegistrarInterface {
         var label = keccak256(name);
         var domain = domains[label];
 
-        if (keccak256(domain.name) != label) {
-            // New listing
-            domain.name = name;
+        if (domain.owner == 0x0) {
```

Comment:
I don't believe this will work, because the function has the `owner_only` modifier, which will return 0 for a newly transferred domain, and thus this code will never execute.

Can you modify `owner` to first check the `domains` map, and if that's 0, then check the deed's previous owner?